phi/tools/firecrawl.py

```diff
@@ -44,33 +62,145 @@ def scrape_website(self, url: str) -> str:
             The results of the scraping.
         """
         if url is None:
-            return "No URL provided"
+            raise ValueError("No URL provided. URL is required for scraping.")
 
         params = {}
         if self.formats:
-            params["formats"] = self.formats
-
+            if set(self.formats).issubset(
+                set(["markdown", "html", "rawHtml", "links", "screenshot", "extract", "screenshot@fullPage"])
+            ):
```

Comment:
If you make this a literal in the class instantiation, Pydantic should handle the check for you?